### PR TITLE
[sw] Remove Make as a SW build system.

### LINKED ISSUE #8: [sw] Move build system to Meson
Replace software Makefiles with Meson build files. The work is broken down as follows:

* [ ] Add support for Meson. Starting with:
   * sw/boot_rom
   * sw/lib
   * sw/host
   * sw/tests
* [ ] Update documentation with new build instructions, e.g.:
  ```console
  $ cd ${REPO_TOP} && ./meson_init.sh
  # Verilator
  $ cd ${REPO_TOP}/build-sim && ninja
  # FPGA
  $ cd ${REPO_TOP}/build-fpga && ninja
  ```
* [ ] Update documentation with new dependencies:
  ```console
  $ pip3 install --user meson
  $ apt-get install ninja-build
  ```
* [ ] Update Docker container to support new dependencies
* [ ] Update otx to support new dependencies
* [ ] Update other build targets, e.g. `sw/benchmark`
* [ ] Update CI to trigger `build-sim` and `build-fpga` builds

### PR DESCRIPTION
**Pull request by mcy**
This change is the culmination of a month or so of work beginning, in commit d258b33, to make all building of OpenTitan software happen with Meson, rather than Make.

Fixes ~~#8~~ and #1032.

Blocked on #1070 and ~~#1117~~.

### CODE REVIEW

**Comment by mcy** [comment]
@asb should be fixed by #1174

**Comment by gkelly** [comment]
> @asb should be fixed by #1174
> 
> Also: is there anywhere I should be announcing this change before it gets approved and merged?

opentitan-tech@lowrisc.org would be a good place to announce.

**Comment by asb** [comment]
@mcy: with the updated patch, you still leave `ci/run_sw_make.sh` in tree (despite removing it from Azure pipelines) - I think this is unintended?

**Comment by mcy** [comment]
@asb Good catch, thanks!

**Comment by mcy** [comment]
See https://github.com/lowRISC/opentitan/issues/1181 for justification on disabling the RISC-V compliance check.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -228,6 +207,9 @@ jobs:
   dependsOn:
     - top_earlgrey_verilator
     - execute_verilated_tests
+  # TODO: re-enable this CI step once RISC-V compliance correctly calls into
+  # Meson.
```
Comment (imphil) [inline azure-pipelines.yml]: Add a bug reference please.

**Pull request merged**